### PR TITLE
Add quantization of multi_gru op and tests

### PR DESCRIPTION
**Pull request by wojtuss**
### PR types
New features

### PR changes
Others

### Describe
This PR enables INT8 quantization of the `multi_gru` op (see PR https://github.com/PaddlePaddle/Paddle/pull/28591).
It also adds a test `save_quant2_model_multi_gru` which transforms a GRU Quant model into GRU INT8 model with `multi_gru` ops.

This PR is a part of the draft PR #28230 and can be merged only after PRs https://github.com/PaddlePaddle/Paddle/pull/28591, https://github.com/PaddlePaddle/Paddle/pull/28601 and https://github.com/PaddlePaddle/Paddle/pull/28604.
It is related to the GRU INT8 model optimization (#27330).

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc [hunk 1]
```diff
@@ -202,23 +206,45 @@ void CPUQuantizePass::DequantizeOutput(Graph* g, Node* op, Node* output,
   if (!scale_attr_name.empty()) op->Op()->SetAttr(scale_attr_name, scale);
 }
 
+bool CPUQuantizePass::AreScalesPresentForVarNames(
+    std::vector<std::string> names) const {
+  auto& scales = Get<VarQuantScale>("quant_var_scales");
+  bool present = true;
+  for (auto name : names) {
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc:214]: Correct me if I'm wrong please, but when **names** param is empty, the function result will be true.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc [hunk 2]
```diff
@@ -202,23 +206,45 @@ void CPUQuantizePass::DequantizeOutput(Graph* g, Node* op, Node* output,
...
+    }
+  }
+  return present;
+}
+
 bool CPUQuantizePass::AreScalesPresentForNodes(
-    const Node* op_node, std::initializer_list<Node*> nodes) const {
+    std::initializer_list<Node*> nodes) const {
   auto& scales = Get<VarQuantScale>("quant_var_scales");
   bool present = true;
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc:226]: ditto with **nodes** param

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc [hunk 3]
```diff
@@ -202,23 +207,45 @@ void CPUQuantizePass::DequantizeOutput(Graph* g, Node* op, Node* output,
...
+      LogScaleIsMissingForVarNode(node);
     }
   }
   return present;
 }
 
+std::pair<bool, LoDTensor> CPUQuantizePass::GetScaleDataByName(
+    const std::string& name) const {
+  auto& scales = Get<VarQuantScale>("quant_var_scales");
+  return scales[name];
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc]: Correct if I'm wrong please, function name suggests getter so maybe **at** function would be better in that case as [] operator has some side effects ?

**Review by arlesniak — CHANGES_REQUESTED** [review]
Thank you for great job Wojtek! Please look at my comments.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc (hunk 1 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc:214]: That's true, and the behavior is OK.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc (hunk 2 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc:226]: True, still OK.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc (hunk 3 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc]: OK, done.

**Review by wozna — DISMISSED** [review]
I checked and now everything looks fine.

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc (hunk 3 above)
Comment (arlesniak) [reply paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc]: OK

**Review by arlesniak — DISMISSED** [review]

**Comment by wojtuss** [comment]
@luotao1 , please, add your review.

**Review by luotao1 — APPROVED** [review]

**Pull request merged**